src/World.cpp

```diff
@@ -2273,19 +2273,69 @@ UInt32 cWorld::SpawnFallingBlock(int a_X, int a_Y, int a_Z, BLOCKTYPE BlockType,
 
 UInt32 cWorld::SpawnExperienceOrb(double a_X, double a_Y, double a_Z, int a_Reward)
 {
+	auto IDs = SpawnSplitExperienceOrbs(a_X, a_Y, a_Z, a_Reward, false);
+	if (IDs.empty())
+	{
+		return cEntity::INVALID_ID;
+	}
+	else
+	{
+		return IDs[0];
+	}
+}
+
+
+
+
+
+
+std::vector<UInt32> cWorld::SpawnSplitExperienceOrbs(double a_X, double a_Y, double a_Z, int a_Reward, bool a_Split)
+{
+	std::vector<UInt32> OrbsID;
+
 	if (a_Reward < 1)
 	{
 		LOGWARNING("%s: Attempting to create an experience orb with non-positive reward!", __FUNCTION__);
-		return cEntity::INVALID_ID;
+		return OrbsID;
 	}
 
-	auto ExpOrb = cpp14::make_unique<cExpOrb>(a_X, a_Y, a_Z, a_Reward);
-	auto ExpOrbPtr = ExpOrb.get();
-	if (!ExpOrbPtr->Initialize(std::move(ExpOrb), *this))
+	std::vector<int> Orbs;
+
+	if (a_Split)
 	{
-		return cEntity::INVALID_ID;
+		cExpOrb::Splite(a_Reward, Orbs);
 	}
-	return ExpOrbPtr->GetUniqueID();
+	else
+	{
+		Orbs.push_back(a_Reward);
+	}
+
+	// Check generate number to decide speed limit (distribute range)
+	int SpeedLimit = (Orbs.size() / 2) + 5;
+	if (SpeedLimit > 10)
+	{
+		SpeedLimit = 10;
+	}
+
+	SpeedLimit *= 100;
+	auto & Random = GetRandomProvider();
+
+	for (auto Element : Orbs)
+	{
+		float SpeedX = Random.RandInt(-SpeedLimit, SpeedLimit) / 100.0f;
+		float SpeedY = Random.RandInt(50) / 100.0f;
+		float SpeedZ = Random.RandInt(-SpeedLimit, SpeedLimit) / 100.0f;
```

Comment:
Is there a reason you're using `RandInt` and converting to float instead of just `RandReal`?

Reply:
sorry I searched and didn't find this method exist